### PR TITLE
[now-build-utils] Add `bypassToken` to `Prerender` for iSSG

### PR DESCRIPTION
**Pull request by Timer**
This pull request adds a new `bypassToken` field to the `Prerender` output.

The field is required to be at least 128 bits for security reasons!

### CODE REVIEW

File: packages/now-build-utils/src/prerender.ts [hunk 1]
```diff
@@ -32,6 +40,22 @@ export class Prerender {
...
+        // Enforce 128 bits of entropy for safety reasons (UUIDv4 size)
+        throw new Error(
+          'The `bypassToken` argument for `Prerender` must be 32 characters or more.'
+        );
+      }
+      this.bypassToken = bypassToken;
+    } else {
+      throw new Error(
+        'The `bypassToken` argument for `Prerender` must be a `string`.'
+      );
```
Comment (styfle) [inline packages/now-build-utils/src/prerender.ts:56]: Your comment on line 11 says that its optional and can therefore be undefined. So why throw here when undefined?

File: packages/now-build-utils/src/prerender.ts [hunk 2]
```diff
@@ -32,6 +40,22 @@ export class Prerender {
     }
     this.group = group;
 
+    if (bypassToken == null) {
```
Comment (styfle) [inline packages/now-build-utils/src/prerender.ts:43]:
```suggestion
    if (!bypassToken) {
```

Ok I see now that you are covering both `null` and `undefined` here.

What do you think of this change?

Comment (Timer) [reply packages/now-build-utils/src/prerender.ts:43]: This captures both `null` and `undefined`. Using only `!` will permit an empty string, which we do not want.

File: packages/now-build-utils/src/prerender.ts (hunk 1 above)
Comment (Timer) [reply packages/now-build-utils/src/prerender.ts:56]: Undefined and null are already handled by this line. This catches non-string values (number, boolean).

File: packages/now-build-utils/src/prerender.ts (hunk 2 above)
Comment (styfle) [reply packages/now-build-utils/src/prerender.ts:43]: If you check `typeof bypassToken === 'string'` and then `!bypassToken` after then it wont allow empty string :smart:

Comment (Timer) [reply packages/now-build-utils/src/prerender.ts:43]: Not sure I understand what change you're proposing. The if clause would just be moving.

**Review by juancampa — APPROVED** [review]

**Review by styfle — APPROVED** [review]

**Pull request merged**